src/header/header.scss

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../user-icon/index';
+@import '../icon/index';
+
+
+@mixin iui-header {
+  @include iui-reset;
+  @include iui-font-family;
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  height: $iui-baseline * 5;
+  transition: height $iui-speed-fast ease-out;
+  @include themed {
+    background-color: t(iui-color-background-1);
+    border-bottom: 2px solid t(iui-color-background-5);
+    color: t(iui-text-color);
+  }
+
+  > .iui-left,
+  > .iui-center,
+  > .iui-right {
+    display: flex;
+    align-items: center;
+    height: 100%;
+    width: 33%;
+  }
+
+  > .iui-left {
+    > .iui-application,
```

Comment:
The fixed 33% width makes the left elements shrink even if there is enough space to actually display them, I wonder if (given that the center div is simply not there if it's empty) something like that could help, keeping the right part at 33%, or use the same logic so one actually have priority on the other ? (Which one should always be visible, regardless of space ?)

```suggestion
    &:nth-last-child(2) {
      width: 66%;
    }
    
    > .iui-application,
```

Reply:
After testing, a better change (again, it forces the iui-center div to be completely removed from the html if it is to be empty) would be to unset the width, for `iui-left` and `iui-right`, and set `iui-left` `overflow` to `hidden`, this is the result, along with some overflow fixes, I can commit to this branch if you want.:
![PwWQ7JfD21](https://user-images.githubusercontent.com/1904889/115886928-69266f00-a41f-11eb-9c60-64a010a1b292.gif)